src/woo_search/search_index/api/viewsets/search.py

```diff
@@ -0,0 +1,112 @@
+from django.utils.translation import gettext_lazy as _
+
+from drf_spectacular.utils import extend_schema
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from woo_search.search_index.api.serializers import (
+    SearchResponseSerializer,
+    SearchSerializer,
+)
+
+from ...client import get_client
+from ...constants import SortChoices
+from ...typing import SearchType
+
+
+@extend_schema(
+    tags=["index"],
+    summary=_("Search"),
+    description=_("Search the publication and document records."),
+    responses={
+        200: SearchResponseSerializer(many=True),
+    },
+)
+class SearchView(APIView):
+    serializer_class = SearchSerializer
+    response_serializer_class = SearchResponseSerializer
+
+    def _query_multi_match(self, query: str):
```

Comment:
1. this is only used in one place, so I fail to see the benefit of defining a static method
2. the `self` argument is not used, so it would make it a `@static_method`, at which point you might just use a regular function definition instead

TBH, I'd rather implement the domain specific methods on our own client class/helper defined in `client.py` to abstract our search parameter translation into ES search operation. That will make it easier to swap out ES in a single place, or handle future breaking changes in elastic search.

With proper code decoupling we can split our domains (input validation / API endpoint definition vs. doing the actual search operation)